### PR TITLE
fuzz: rework df_exec_cmd_check()

### PR DESCRIPTION
**Pull request by mrc0mmand**
Explicitly use fork() & execl() to avoid the file descriptor
shenanigans, and move & rename the function to util.[ch].

Also, when at it, add a unit test for the function as well.

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -60,3 +66,52 @@ int safe_strtoull(const gchar *p, guint64 *ret)
...
+                        break;
+                }
+
+                return status.si_status;
+        }
+
+        /* Child process */
+        g_auto(fd_t) null_fd = -1;
+
+        if (df_get_log_level() < DF_LOG_LEVEL_VERBOSE) {
```
Comment (evverx) [inline src/util.c:101]: I don't think the log level of dfuzzer should affect the output of the command. Its stdin/stdout/stderr should probably be redirected to `/dev/null` by default and controlled by a separate command line switch that could be set independently.

**Comment by evverx** [comment]
@mrc0mmand could you move the first commit (with `g_assert`) to a separate PR? I think it should be good to go.

**Comment by mrc0mmand** [comment]
> @mrc0mmand could you move the first commit (with `g_assert`) to a separate PR? I think it should be good to go.

Sure thing, done in https://github.com/dbus-fuzzer/dfuzzer/pull/121.

**Comment by evverx** [comment]
> now we can do --debug or --debug=general for the expected/current behavior, --debug=command to show only output from the --command

I'm kind of on the fence about this. On the one hand it should make it possible to extend that part easily but on the other hand as a user I'd prefer to pass something like `-e '/some/weird/script' --keep-command-output'` and be done with it.

**Comment by mrc0mmand** [comment]
> > now we can do --debug or --debug=general for the expected/current behavior, --debug=command to show only output from the --command
> 
> I'm kind of on the fence about this. On the one hand it should make it possible to extend that part easily but on the other hand as a user I'd prefer to pass something like `-e '/some/weird/script' --keep-command-output'` and be done with it.

I don't really have a preference here, so I can definitely leave the `--debug` option as is and just add a new one :-)

**Comment by evverx** [comment]
Just to clarify, I'm just used to tools where I have to flip a switch like `-Q` in `honggfuzz` or `AFL_DEBUG_CHILD` in `afl-fuzz` to start capturing the output of various binaries so whatever I'm saying here is subjective and can be ignored :-)

**Comment by evverx** [comment]
> I can definitely leave the --debug option as is and just add a new one :-)

I think I'd go with this. If at some point it causes any issues in terms of extending the logger it will be on me :-)

**Comment by evverx** [comment]
I think I should have read the commit message carefully. `fork` followed up by `execl` lets `dfuzzer` avoid saving/restoring its stdin/stdout/stderr with `dup/dup2`. LGTM. I'll go ahead and merge it. Thanks!

**Pull request merged**